internal/datastore/memdb/caveat.go

```diff
@@ -36,31 +32,8 @@ func (r *memdbReader) ReadCaveatByName(name string) (*core.Caveat, error) {
 	return r.readUnwrappedCaveatByName(tx, name)
 }
 
-func (r *memdbReader) ReadCaveatByID(ID datastore.CaveatID) (*core.Caveat, error) {
-	r.lockOrPanic()
-	defer r.Unlock()
-
-	tx, err := r.txSource()
-	if err != nil {
-		return nil, err
-	}
-	return r.readCaveatByID(tx, ID)
-}
-
-func (r *memdbReader) readCaveatByID(tx *memdb.Txn, ID datastore.CaveatID) (*core.Caveat, error) {
-	found, err := tx.First(tableCaveats, indexID, ID)
-	if err != nil {
-		return nil, err
-	}
-	if found == nil {
-		return nil, datastore.NewCaveatIDNotFoundErr(ID)
-	}
-	c := found.(*caveat)
-	return c.Unwrap(), nil
-}
-
 func (r *memdbReader) readCaveatByName(tx *memdb.Txn, name string) (*caveat, error) {
-	found, err := tx.First(tableCaveats, indexName, name)
+	found, err := tx.First(tableCaveats, indexID, name)
```

Comment:
shouldn't the index be `indexName`?

Reply:
That's not the convention used in the codebase. It uses `indexID` because it's the "ID" column for `hashicorp/go-memdb`, which is required. You can think of it as the "PK" constraint

Reply:
That's not confusing at all! /s